investigations/Possible_Instrument_View_Improvements.md

```diff
@@ -0,0 +1,48 @@
+# Investigation of Possible Instrument 2.0 Effects on Instrument View Performance.
+This document outlines areas in which the performance of the instrument view may benefit from the introduction of the new Instrument 2.0 caching layers. Tests were carried out using the Microsoft Visual Studio debugger and profiling tools which provided information on which Instrument 1.0 methods are most heavily used when interacting with the instrument view.
+
+### Parametrised Instrument Allocation
+In several scenarios within the Instrument View (loading, switching between unwrapped views and picking) a parametrised version of the instrument is created before attempting to access detectors. Every call to `ExperimentInfo::getInstrument()` results in a call to `ExperimentInfo::makepParameterizedInstrument()`, which in turn is triggered by every call to `InstrumentActor::getInstrument()` or `ComponentActor::getComponent()`. For SANS2D, there are 255 calls to the parametrised instrument constructor, for GEM there are ~130K. Similar figures were recorded for switching instrument representations (unwrapped views). This seems to be, in general terms, a design flaw in Instrument 1.0. However, Instrument 2.0 obviates the need to access parameter maps or even the creation of a parametrised instrument. It therefore follows that any access which replaces this object creation with a table lookup could drastically improve performance.
+
+### Caching Layer Lookup vs Detector Object Access
+Following on from the last point, in most places where information about individual detectors are required, a pointer/reference to the detector object is obtained in order to retrieve detector id, shape, bounding boxes and corresponding workspace data. Instrument 2.0 simplifies this access by making most of this information directly accessible from the `DetectorInfo`/`ComponentInfo`. 
+
+### List of Operations Instrument View requires of Instrument 1.0
+- Parametrised instrument constructor 
+- `Instrument::getDetector()`
+- `Instrument::getSample()`
+- `Instrument::getSource()`
+- `Instrument::getDetectorIDs()`
+- `Instrument::getBoundingBox()`
+- `Instrument::getPhysicalInstrument()`
+- `Instrument::setPhysicalInstrument()`
+- `Detector::getRotation()`
+- `Detector::getPos()`
+- `Detector::getID()`
+- `Component::getParent()`
+- `Component::getComponentID()`
+- `ObjComponent::getBoundingBox()`
+- `ObjComponent::shape()`
+- `ObjComponent::draw()`
```

Comment:
If you have the information already. Would be nice to have this ordered by relative footprint (cost), which should probably take into account the number of calls to these methods in a typical instrument. Add that if you happen to have that information still.